contracts/lst-staking/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![no_std]
+
+multiversx_sc::imports!();
+
+mod esdt;
+use esdt::{AvailableAmount, Esdt};
+
+#[multiversx_sc::contract]
+pub trait LstStakingContract {
+    #[init]
+    fn init(&self, unbond_period: u64) {
+        self.unbond_period().set(unbond_period);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn whitelist_token(&self, token: TokenIdentifier) {
+        let _ = self.token_whitelist().insert(token);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn blacklist_token(&self, token: &TokenIdentifier) {
+        let _ = self.token_whitelist().swap_remove(token);
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn stake(&self) {
+        let payments = self.call_value().all_esdt_transfers();
+        let caller = self.blockchain().get_caller();
+        for payment in payments.iter() {
+            self.require_whitelisted_token(&payment.token_identifier);
```

Comment:
add a check for token to be valid esdt token and nonce = 0 if we accept only esdt payments